bids2table/entities.py

```diff
@@ -257,7 +222,34 @@ def with_update(
             data.update(entitities)
         if kwargs:
             data.update(kwargs)
-        return BIDSEntities.from_dict(data)
+        return self.__class__.from_dict(data)
+
+
+def make_bids_field(
+    entity_schema: Dict[str, Any],
+) -> Tuple[str, Any, Any]:
+    """
+    BIDS entity dataclass field.
+    """
+    metadata = {
+        "name": entity_schema["name"],
+        "display_name": entity_schema["display_name"],
+        "allowed_values": entity_schema.get("enum"),
+    }
+    type_ = {"integer": int}.get(entity_schema["type"], str)
+    field_ = field(default=None, metadata=metadata)
+    return entity_schema["name"], Optional[type_], field_
+
+
+BIDSEntities: Any = make_dataclass(
```

Comment:
```suggestion
BIDSEntities: type = make_dataclass(
```

Reply:
I had it like this, but mypy does not like it at all (results in everything being flagged missing-attr). There is an open issue: https://github.com/python/mypy/issues/6063

Reply:
Cool...